File: python_examples/demos/tiktok_video.py
#!/usr/bin/env python3
"""TikTok video creation demo - find video and create short clip."""

import asyncio
import os,re
from dotenv import load_dotenv
from mix_python_sdk import Mix
from mix_python_sdk.helpers import send_with_callbacks
from mix_python_sdk.tool_models import MediaShowcaseParams, CoreToolName


async def main():
    load_dotenv()

    async with Mix(server_url=os.getenv("MIX_SERVER_URL")) as mix:
        mix.preferences.update_preferences(
            preferred_provider="anthropic",
            main_agent_model="claude-sonnet-4-5",
        )

        session = mix.sessions.create(title="TikTok Video Demo")

        def handle_tool(t):
            if t.name != CoreToolName.SHOW_MEDIA: return
            if not t.input or (isinstance(t.input, str) and not t.input.strip()): return

            try:
                params = MediaShowcaseParams.model_validate_json(t.input) if isinstance(t.input, str) else MediaShowcaseParams.model_validate(t.input)
                for o in params.outputs:
                    print(f"\n{o.title}\n   {o.description or ''}\n   {o.path or ''}\n")
            except Exception:
                pass  # Ignore parsing errors for incomplete tool calls

        await send_with_callbacks(
            mix,
            session_id=session.id,
            message="Find the top cat video and create a 5 sec tiktok video from it. Add a title animation. Export it and show.",
            on_tool=handle_tool,
            on_content=lambda text: print(re.sub(r'(\.)([A-Z])', r'\1\n\2', text), end="", flush=True),
            on_error=lambda error: print(f"\n❌ {error}"),

        )


if __name__ == "__main__":
    asyncio.run(main())

File: typescript_examples/demos/tiktok_video.ts
#!/usr/bin/env tsx
/**
 * TikTok video creation demo - find video and create short clip.
 */

import { config } from "dotenv";
import { Mix } from "mix-typescript-sdk";
import { sendWithCallbacks } from "../helpers.js";

config();

async function main() {
  const mix = new Mix({
    serverURL: process.env.MIX_SERVER_URL,
  });

  // Update preferences
  await mix.preferences.update({
    preferred_provider: "anthropic",
    main_agent_model: "claude-sonnet-4-5",
  });

  // Create session
  const session = await mix.sessions.create({
    title: "TikTok Video Demo",
  });

  // Handle tool events
  function handleTool(tool: any) {
    if (tool.name !== "show_media" || !tool.input) return;

    try {
      const input = typeof tool.input === "string" ? JSON.parse(tool.input) : tool.input;
      const outputs = input.outputs || [];

      for (const output of outputs) {
        console.log(`\n${output.title}`);
        if (output.description) console.log(`   ${output.description}`);
        if (output.path) console.log(`   ${output.path}`);
        console.log();
      }
    } catch (e) {
      // Silently ignore parse errors
    }
  }

  // Create TikTok video
  await sendWithCallbacks(
    mix,
    session.id,
    "Find the top cat video and create a 5 sec tiktok video from it. Add a title animation. Export it and show.",
    {
      onTool: handleTool,
      onContent: (text) => {
        // Add line breaks after periods before capital letters for readability
        const formatted = text.replace(/(\.)([A-Z])/g, "$1\n$2");
        process.stdout.write(formatted);
      },
      onError: (error) => console.log(`\n❌ ${error}`),
    }
  );
}

main()
  .catch(console.error)
  .finally(() => process.exit(0));
